tasks/run_puppet_device.rb

```diff
@@ -79,7 +80,7 @@ def run_puppet_device(devices, noop, timeout)
     result = ''
 
     begin
-      Open3.popen2e(puppet_command, 'device', user, '--waitforcert=0', '--verbose', target, noop) do |_, oe, w|
+      Open3.popen2e(puppet_command, 'device', user, '--waitforcert=0', apply, '--verbose', target, noop) do |_, oe, w|
```

Comment:
Nit: Could you move the optional `apply` variable out from between the two `--waitforcert=0` and `--verbose' options to the end of the list of options?

```
      Open3.popen2e(puppet_command, 'device', user, '--waitforcert=0', '--verbose', target, noop, apply) do |_, oe, w|
```